catroid/src/test/java/org/catrobat/catroid/test/merge/MergeProjectTest.java

```diff
@@ -0,0 +1,96 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.merge;
+
+import android.content.Context;
+
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.formulaeditor.UserList;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.catrobat.catroid.io.XstreamSerializer;
+import org.catrobat.catroid.merge.MergeProject;
+import org.catrobat.catroid.test.MockUtil;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertNull;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({XstreamSerializer.class})
+public class MergeProjectTest {
+	private Project firstProject;
+	private Project secondProject;
+	private MergeProject mergeTask;
+
+	@Before
+	public void setUp() {
+		Context mockContext = MockUtil.mockContextForProject();
+		firstProject = new Project(mockContext, "testProjectOne");
+		secondProject = new Project(mockContext, "testProjectTwo");
+
+		firstProject.addUserList(new UserList("userListOne"));
+		firstProject.addUserVariable(new UserVariable("userVariableOne"));
+		firstProject.addScene(new Scene("sceneOne", firstProject));
+
+		secondProject.addUserList(new UserList("userListTwo"));
+		secondProject.addUserVariable(new UserVariable("userVariableTwo"));
+		secondProject.addScene(new Scene("sceneTwo", secondProject));
+
+		mergeTask = new MergeProject(firstProject, secondProject, mockContext, false);
+
+		PowerMockito.mockStatic(XstreamSerializer.class);
+		XstreamSerializer xstreamSerializerMock = PowerMockito.mock(XstreamSerializer.class);
+		PowerMockito.when(XstreamSerializer.getInstance()).thenReturn(xstreamSerializerMock);
+		PowerMockito.when(xstreamSerializerMock.saveProject(Mockito.isA(Project.class))).thenReturn(true);
+	}
+
+	@Test
+	public void mergeTaskNoConflictTest() {
+		assertNotNull(mergeTask.performMerge("mergedProject"));
+	}
+
+	@Test
+	public void mergeTaskWithSceneConflictTest() {
+		firstProject.addScene(new Scene("sceneTwo", secondProject));
+		assertNull(mergeTask.performMerge("mergedProject"));
+	}
+
+	@Test
+	public void mergeTaskWithUserListConflictTest() {
+		firstProject.addUserList(new UserList("userListTwo"));
+		assertNull(mergeTask.performMerge("mergedProject"));
+	}
+
+	@Test
+	public void mergeTaskWithUserVariableConflictTest() {
+		firstProject.addUserVariable(new UserVariable("userVariableOne"));
```

Comment:
strings that you use to deliberately test for colissions should *not* be hardcoded, but constants.